File: Python/day7.py
#Write a program to find Number of days in a given month of a given year
def yrchk(year):
    if (year % 400 == 0) or (year % 100 != 0 and year % 4 == 0):
        return True
    return False
    
def daizz(month, year):
    mon31 = ["January", "March", "May", "July", "August", "October", "December"]
    mon30 = ["April", "June", "September", "November"]
    month = month.capitalize()
    if month in mon31:
        return 31
    elif month in mon30:
        return 30
    elif month == "February":
        return 29 if yrchk(year) else 28
    else:
        return "Invalid month" 

month = input("Enter the name of the month: ")
year = int(input("Enter the year: "))
days = daizz(month, year)
print(f"{month} {year} has {days} days.")

File: Java/day7.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter month: ");
        String month = sc.nextLine(); 
        System.out.print("Enter year: ");
        int year = sc.nextInt(); 
        switch(month.toLowerCase()) {
            case "january": case "march": case "may": case "july":
            case "august": case "october": case "december":
                System.out.println(month + " " + year + " has 31 days.");
                break;
            case "april": case "june": case "september": case "november":
                System.out.println(month + " " + year + " has 30 days.");
                break;
            case "february":
                if (isLeapYear(year)) {
                    System.out.println("February " + year + " has 29 days.");
                } else {
                    System.out.println("February " + year + " has 28 days.");
                }
                break;
            default:
                System.out.println("Invalid month.");
        }
        
        sc.close();
    }
    public static boolean isLeapYear(int year) {
        return (year % 400 == 0) || (year % 100 != 0 && year % 4 == 0);
    }
}
